jinahub/indexers/searcher/FaissSearcher/faiss_searcher.py

```diff
@@ -91,49 +111,62 @@ def __init__(
         self._prefetch_data = []
 
         self.logger = get_logger(self)
+        self._load_dump(dump_path, dump_func, prefetch_size, **kwargs)
+
+    def _load_dump(self, dump_path, dump_func, prefetch_size, **kwargs):
+        dump_path = dump_path or kwargs.get('runtime_args', {}).get('dump_path')
+
+        iterator = None
 
-        dump_path = dump_path or kwargs.get('runtime_args').get('dump_path')
         if dump_path is not None:
             self.logger.info(
                 f'Start building "FaissIndexer" from dump data {dump_path}'
             )
             ids_iter, vecs_iter = import_vectors(
                 dump_path, str(self.runtime_args.pea_id)
             )
+            iterator = zip(ids_iter, vecs_iter)
+        elif dump_func is not None:
+            iterator = dump_func(shard_id=self.runtime_args.pea_id)
+        else:
+            self.logger.warning(
+                'No "dump_data" or "dump_func" passed to "FaissIndexer".'
+                ' Use .rolling_update() to re-initialize it...'
+            )
+            return
 
-            self._doc_ids = np.array(list(ids_iter))
-            self._doc_id_to_offset = {v: i for i, v in enumerate(self._doc_ids)}
-
+        if iterator is not None:
+            iterator = self._iterate_vectors_and_save_ids(iterator)
+            self._prefetch_data = []
             if self.prefetch_size and self.prefetch_size > 0:
                 for _ in range(prefetch_size):
                     try:
-                        self._prefetch_data.append(next(vecs_iter))
-
+                        self._prefetch_data.append(next(iterator))
                     except StopIteration:
                         break
             else:
-                self._prefetch_data = list(zip(ids_iter, vecs_iter))
+                self._prefetch_data = list(iterator)
```

Comment:
Why is this prefetching happening here anyhow? The only reason is to get `self.num_dim` and `self.dtype` filled. This seems a quite weak reason to do so. This can happen in the very first `next` call inside `_iterate_vectors_and_save_ids` now. This is much cleaner.

Reply:
Yes, idk. @numb3r3 ?

Reply:
Can be addressed in a future PR maybe.